### PR TITLE
Deprecated SFAuthenticationManager

### PR DESCRIPTION
**Pull request by trooper2013**
This PR takes care of deprecating SFAuthenticationManager. I have tested AccountEditor, SmartSyncExplorer, RestAPIExplorer & IDP situations as well. All tests were running locally except for one [SmartSyncTestSuite testFetchSObjects]. Its expecting 2 objects but gets three. Needs more scrutiny.

- There is still code in the SDK that needs to use the component if useLegacyAuthentication has been set. I have added 2 simple macros to SUPPRESS warnings.  (SFSDK_USE_DEPRECATED_BEGIN , SFSDK_USE_DEPRECATED_END)

- A few blocks of code where SFAuthenticationManager was being used was easily swapped out for SFUserAccountManager had to do with reading mostly properties that are being set into the SFUserAccountManager anyway.

- Tests that use  SFAuthenticationManager will continue to do so until we do away with it.  I've added the same macros to these blocks to suppress warnings.

Some more  refactoring efforts were needed.
- Additionally added moved resetCookie to the SFSDKWebViewStateManager.
- Missing properties in SFUserAccountManager.
- Hybrid plugin was upgraded to use SFUserAccountManager.
- Code using coordinator to fetch credentials have been swapped out  to use current user's credentials instead.
- SalesforceSDKManager was implementing SFAuthenticationManager delegate in its public header.  Now has been moved to private.

### CODE REVIEW

File: native/SampleApps/RestAPIExplorer/RestAPIExplorer/Classes/AppDelegate.m [hunk 1]
```diff
@@ -48,6 +48,7 @@ - (id)init
     self = [super init];
     if (self) {
         [SFAuthenticationManager sharedManager].advancedAuthConfiguration = SFOAuthAdvancedAuthConfigurationRequire;
```
Comment (wmathurin) [inline native/SampleApps/RestAPIExplorer/RestAPIExplorer/Classes/AppDelegate.m]: Shouldn't that line go?

File: libs/SmartStore/SmartStoreTests/SFMultipleSmartStoresTests.m [hunk 2]
```diff
@@ -27,7 +27,7 @@
  */
 
 #import "SFMultipleSmartStoresTest.h"
-
+#import <SalesforceSDKCore/SFAuthenticationmanager.h>
```
Comment (wmathurin) [inline libs/SmartStore/SmartStoreTests/SFMultipleSmartStoresTests.m]: Why do we need that import?

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Test/TestSetupUtils.m [hunk 3]
```diff
@@ -89,14 +89,15 @@ + (SFSDKTestCredentialsData *)populateAuthCredentialsFromConfigFileForClass:(Cla
...
+SFSDK_USE_DEPRECATED_BEGIN
 + (void)synchronousAuthRefresh
 {
     // All of the setup and validation of prerequisite auth state is done in populateAuthCredentialsFromConfigFile.
     // Make sure that method has run before this one.
     NSAssert(credentials!=nil, @"You must call populateAuthCredentialsFromConfigFileForClass before synchronousAuthRefresh");
     __block SFSDKTestRequestListener *authListener = [[SFSDKTestRequestListener alloc] init];
     __block SFUserAccount *user = nil;
+
     [[SFAuthenticationManager sharedManager]
```
Comment (wmathurin) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Test/TestSetupUtils.m]: Why are we not switching to SFUserAccountManager here?

**Comment by brandonpage** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Error">
          1 Error
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:no_entry_sign:</td>
      <td data-sticky="false">Tests have failed, see below for more information.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">Needs testing on a Phone if change is non-trivial</td>
    </tr>
  </tbody>
</table>

### Tests: 

Classname | Name|
--- | ---|
SalesforceRestAPITests | testBlockUpdate|
SalesforceRestAPITests | testBlockUpdate|
SalesforceRestAPITests | testUpsert|
SalesforceRestAPITests | testUpsert|
SmartSyncTestSuite | testFetchApexRestObjectsFromServer|
SmartSyncTestSuite | testFetchSObjects|
SmartSyncTestSuite | testSyncApexRestObjectWithServerCreate|
SmartSyncTestSuite | testSyncApexRestObjectWithServerDelete|
SmartSyncTestSuite | testSyncApexRestObjectWithServerRead|
SmartSyncTestSuite | testSyncApexRestObjectWithServerUpdate|


## Current coverage for SalesforceAnalytics.framework is `66.64%`
✅ *No files affecting coverage found*

---
## Current coverage for SalesforceHybridSDK.framework is `43.78%`
Files changed | - | - 
--- | --- | ---
SFAccountManagerPlugin.m | `0.00%` | :skull:
SFAccountManagerPlugin.m | `0.00%` | :skull:
SalesforceOAuthPlugin.m | `18.84%` | :skull:
SalesforceOAuthPlugin.m | `18.84%` | :skull:
SFHybridViewController.m | `23.38%` | :skull:
SFHybridViewController.m | `23.38%` | :skull:

---
## Current coverage for SalesforceSDKCore.framework is `50.38%`
Files changed | - | - 
--- | --- | ---
SFSDKLoginHostListViewController.m | `0.00%` | :skull:
SFDefaultUserManagementViewController.m | `0.00%` | :skull:
SFSDKLoginHostListViewController.m | `0.00%` | :skull:
SFDefaultUserManagementDetailViewController.m | `0.00%` | :skull:
SFDefaultUserManagementViewController.m | `0.00%` | :skull:
SFDefaultUserManagementDetailViewController.m | `0.00%` | :skull:
SFLoginViewController.m | `17.93%` | :skull:
SFLoginViewController.m | `17.93%` | :skull:
SFSDKWebViewStateManager.m | `22.12%` | :skull:
SFSDKWebViewStateManager.m | `22.12%` | :skull:
SFPushNotificationManager.m | `26.50%` | :no_entry_sign:
SFPushNotificationManager.m | `26.50%` | :no_entry_sign:
SFAuthenticationManager.m | `34.59%` | :no_entry_sign:
SFAuthenticationManager.m | `34.59%` | :no_entry_sign:
SFSDKSalesforceAnalyticsManager.m | `34.70%` | :no_entry_sign:
SFSDKSalesforceAnalyticsManager.m | `34.70%` | :no_entry_sign:
SFSecurityLockout.m | `41.63%` | :no_entry_sign:
SFSecurityLockout.m | `41.63%` | :no_entry_sign:
SFSDKTestRequestListener.m | `50.93%` | :warning:
SFSDKTestRequestListener.m | `50.93%` | :warning:
SalesforceSDKManager.m | `58.59%` | :warning:
SalesforceSDKManager.m | `58.59%` | :warning:
SFOAuthSessionRefresher.m | `77.60%` | :warning:
SFOAuthSessionRefresher.m | `77.60%` | :warning:
TestSetupUtils.m | `79.22%` | :warning:
TestSetupUtils.m | `79.22%` | :warning:
SFRestAPI.m | `88.43%` | :white_check_mark:
SFRestAPI.m | `88.43%` | :white_check_mark:

---
## Current coverage for SmartStore.framework is `64.64%`
Files changed | - | - 
--- | --- | ---
SalesforceSDKManagerWithSmartStore.m | `69.23%` | :warning:
SalesforceSDKManagerWithSmartStore.m | `69.23%` | :warning:

---
## Current coverage for SmartSync.framework is `79.50%`
Files changed | - | - 
--- | --- | ---
SFSmartSyncMetadataManager.m | `55.35%` | :warning:
SFSmartSyncMetadataManager.m | `55.35%` | :warning:
SFSmartSyncCacheManager.m | `71.17%` | :warning:
SFSmartSyncCacheManager.m | `71.17%` | :warning:
SFSmartSyncSyncManager.m | `89.64%` | :white_check_mark:
SFSmartSyncSyncManager.m | `89.64%` | :white_check_mark:

---

> Powered by [xcov](https://github.com/nakiostudio/xcov)

<p align="right" data-meta="generated_by_CI-result">
  Generated by :no_entry_sign: <a href="http://danger.systems/">Danger</a>
</p>

**Pull request merged**